core/storage/statistics/gae_models.py

```diff
@@ -474,6 +640,44 @@ def get_versions(cls, exploration_id):
             ).fetch(feconf.DEFAULT_QUERY_LIMIT)]
 
 
+class HintViewsModel(base_models.BaseModel):
+    """Stores Hint Statistics information for an exploration.
+
+    The id/key of instanced of this class has the form
+        [EXPLORATION_ID]:[STATE_NAME]:[EXPLORATION_VERSION]
+    """
+    # Explicitly store exploration id, exploration version and state name
+    # so we can easily do queries on them.
+    exploration_id = ndb.StringProperty(indexed=True, required=True)
+    exploration_version = ndb.IntegerProperty(indexed=True, required=True)
+    state_name = ndb.StringProperty(indexed=True, required=True)
+    # List of viewed hints dicts, each of which is stored as JSON blob.
+    # Keyed by state_name[hint_index] that describes the number of views
+    # and number of successful answer attempts on viewing a hint.
+    # {num_views: ..., num_succeeds: ...}
+    viewed_hints_list = ndb.JsonProperty(repeated=True, indexed=True)
+    # Stores the number of solution views for state.
+    solution_views = ndb.IntegerProperty(indexed=True, required=False)
+    # When the statistics was reset last.
```

Comment:
When the statistics were last reset.

Though if you're storing this per-exploration-version, you'll never need to do a reset, right? The stats will be auto-reset when the exploration creator commits a new version.

Reply:
Yep, it gets reset when the creator commits a new version. I'll remove this field?

Reply:
OK.